x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/util/ShapeTestUtils.java

```diff
@@ -96,6 +99,16 @@ public static Polygon randomPolygon(boolean hasAlt) {
         return new Polygon(linearRing(floatsToDoubles(lucenePolygon.getPolyX()), floatsToDoubles(lucenePolygon.getPolyY()), hasAlt));
     }
 
+    static double area(XYPolygon p) {
```

Comment:
This is copied from the geo version, but I wonder (in both geo and cartesian) whether this is really a true 'area'. Perhaps the method should have another name?

Reply:
The method could be private? Feel free to rename it, I think this just the Shoelace formula applied here.